File: plugin.sync.bug/src/main/scala/com/anarsoft/race/detection/model/description/DescVisitor.scala
package com.anarsoft.race.detection.model.description

trait DescVisitor {
  
    
  def visitClassDescription( name : String,source : String , isThreadSafe : Boolean, isStateLess : Boolean, except : Array[String], superName : String, interfaces: Array[String]);
	
  def visitMethodDescription( name : String, id : Int, desc : String,access : Int,lineNumber : Int );
  
 // def visitAnnotationArray(annotationArray: Array[String]);
  
  
  def endMethod();
  
  
	
  def visitFieldAccessDescription( name: String,  owner : String,  id : Int,
			 isStatic : Boolean,  isWrite : Boolean, isTraced : Boolean,
			 isFinal : Boolean);
	def visitFieldDescription( name : String,  id : Int,  desc : String,  access : Int);
  
  
  
}

File: trace-agent-bootstrap/src/main/java/com/anarsoft/trace/agent/description/DescVisitor.java
package com.anarsoft.trace.agent.description;

public interface DescVisitor {
	
	void visitClassDescription(String name);
	void visitMethodDescription(String name,int id,String desc,int access);
	void visitFieldAccessDescription(String name, String owner, int id,
			boolean isStatic, boolean isWrite,boolean isTraced,boolean isFinal);
	void visitFieldDescription(String name, int id, String desc, int access);

}
